src/mlpack/methods/preprocess/preprocess_one_hot_encoding_main.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file methods/preprocess/preprocess_one_hot_encoding_main.cpp
+ * @author Jeffin Sam
+ *
+ * A IO executable to do One-Hot Encoding on features from a dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/io.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/one_hot_encoding.hpp>
+
+PROGRAM_INFO("One Hot Encoding",
+    // Short description.
+    "A utility to do one hot encoding on features of dataset.",
+    // Long description.
+    "This utility takes a dataset and a vector of indices and does one hot "
+    "encoding of the respective features at those indices. Indices represent "
+    "the IDs of the dimensions to be one-hot encoded."
+    "\n\n"
+    "The output matrices with encoded features may be saved with the " +
+    PRINT_PARAM_STRING("output") + " parameters."
+    "\n\n"
+    "So, a simple example where we want to encode 1st and 3rd feature"
+    " from dataset " + PRINT_DATASET("X") + " into " +
+    PRINT_DATASET("X_output") + " would be"
+    "\n\n" +
+    PRINT_CALL("preprocess_one_hot_encoding", "input", "X", "output",
+        "X_ouput", "dimensions", 1 , "dimensions", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save one hot encoded features "
+    "data to.", "o");
+
+PARAM_VECTOR_IN_REQ(int, "dimensions", "Index of dimensions that"
+    "need to be one hot encoded.", "d");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Load the data.
+  const arma::mat& data = IO::GetParam<arma::mat>("input");
+  vector<int>& indices =
+    IO::GetParam<vector<int> >("dimensions");
+  vector<size_t>copyIndices(indices.size());
+  for (size_t i = 0; i < indices.size(); i++)
+  {
+    if (indices[i] < 0 || indices[i] >= data.n_rows)
+      throw std::runtime_error("Indices should be having positive value"
+      " and less than then the number of dimesnions of the input data");
+    copyIndices[i] = (size_t)indices[i];
+  }
```

Comment:
We can actually use the code in `src/mlpack/core/data/param_checks.hpp` for this.  Maybe this would work:

```suggestion
  RequireParamValue<std::vector<int>>("dimensions", [data](std::vector<int>& x) {
        for (dim : x)
          if (dim < 0 || dim > data.n_rows)
            return false;
        return true;
      }, true, "dimensions must be greater than 0 and less than the number of dimensions");
```
I am not sure that will work exactly---you might need to debug my lambda function?  Anyway, give it a shot, see what you think. :)

Reply:
Not sure what the error is
<img width="1472" alt="Screenshot 2020-08-19 at 9 41 22 PM" src="https://user-images.githubusercontent.com/21286134/90661587-0152cf80-e265-11ea-8733-cd5b383fa7be.png">

Reply:
Does it work if the lambda signature is `[data](std::vector<int> x)`?  That will incur a copy of the int vector, but, it certainly won't be a very long vector, so maybe it won't be too painful.  Also, you could try `[data](const std::vector<int>& x)` too.